prov/efa/src/efa_user_info.c

```diff
@@ -269,34 +269,133 @@ int efa_user_info_get_dgram(uint32_t version, const char *node, const char *serv
 	return -FI_ENODATA;
 }
 
+#if HAVE_CUDA || HAVE_NEURON || HAVE_SYNAPSEAI
+/**
+ * @brief determine if EFA provider should claim support of FI_HMEM in info
+ * @param[in]	version		libfabric API version used by user
+ * @param[in]	hints		hints from user's call to fi_getinfo()
+ * @return	true, if EFA provider should claim support of FI_HMEM
+ * 		false, otherwise
+ */
+bool efa_user_info_should_support_hmem(int version, const struct fi_info *hints)
+{
+	/*
+	 * FI_HMEM is a primary capability. If user did not explicitly request
+	 * it, we should not cliam support of it.
+	 * only when user explicitly requeisted it.
+	 */
+	if (!hints || !(hints->caps & FI_HMEM))
+		return false;
+
+	/* Note that the default behavior of EFA provider is different between
+	 * libfabric API version.
+	 *
+	 * For libfabric API version 1.17 and earlier, EFA provider does not
+	 * support FI_HMEM unless GPUDirect RDMA is available.
+	 *
+	 * For libfabric API version 1.18 and later, EFA provider will claim support
+	 * of FI_HMEM as long as CUDA library is initialized and a CUDA device is
+	 * available. On an system without GPUDirect RDMA, the support of CUDA memory
+	 * is implemented by calling CUDA library. If user does not want EFA provider
+	 * to use CUDA library, the user can call use fi_setopt to set
+	 * FI_OPT_CUDA_API_PERMITTED to false. On an system without GPUDirect RDMA,
+	 * the call would fail.
+	 */
+	if (FI_VERSION_GE(version, FI_VERSION(1, 18)) && hmem_ops[FI_HMEM_CUDA].initialized) {
+		EFA_INFO(FI_LOG_CORE,
+			"User is using 1.18 API and CUDA library and device are availble, cliam support of HMEME.");
```

Comment:
`API version >= 1.18 and CUDA library and device are available. Claim support of HMEM`

Reply:
No. it must be "user is using 1.18 API" because user can chose different version of API.

Reply:
Typo in "cliam" and "HMEME".

Reply:
> No. it must be "user is using 1.18 API" because user can chose different version of API.

Should it be `user is using >= 1.18 API`?

Reply:
Fixed both in new revision